src/converter.ts

```diff
@@ -0,0 +1,127 @@
+import React, { type CSSProperties } from 'react'
+
+import type { AbstractElement } from '@fortawesome/fontawesome-svg-core'
+
+import type { FontAwesomeIconProps } from './types/icon-props'
+import { camelize } from './utils/camelize'
+
+function capitalize(val: string): string {
+  return val.charAt(0).toUpperCase() + val.slice(1)
+}
+
+// Cache for parsed styles to avoid re-parsing identical style strings
+// This is beneficial for performance not only due to the fact this function is called on each React render
+// but also when multiple icons are rendered on a page, since many icon SVGs share similar built-in styles.
+export const styleCache = new Map<string, CSSProperties>()
+const STYLE_CACHE_LIMIT = 1000
+
+function styleToObject(style: string): CSSProperties {
+  // Check cache first
+  if (styleCache.has(style)) {
+    return styleCache.get(style)!
+  }
+
+  const result: Record<string, string> = {}
+
+  let start = 0
+  const len = style.length
+
+  while (start < len) {
+    // Find the next semicolon
+    const semicolonIndex = style.indexOf(';', start)
+    const end = semicolonIndex === -1 ? len : semicolonIndex
+
+    // Extract the property-value pair
+    const pair = style.slice(start, end).trim()
+    if (pair) {
+      const colonIndex = pair.indexOf(':')
+      if (colonIndex > 0) {
+        const rawProp = pair.slice(0, colonIndex).trim()
+        const value = pair.slice(colonIndex + 1).trim()
+
+        if (rawProp && value) {
+          const prop = camelize(rawProp)
+          result[prop.startsWith('webkit') ? capitalize(prop) : prop] = value
+        }
+      }
+    }
+
+    start = end + 1
+  }
+
+  // Prevent memory leaks by clearing cache when it gets too large
+  if (styleCache.size === STYLE_CACHE_LIMIT) {
+    styleCache.clear()
```

Comment:
It's probably not worth it until there's some kind of feedback from library users, _but_ I immediately thought of an LRU cache when I saw this.

Instead of clearing the whole thing you could do this to remove the oldest entry:

```suggestion
    styleCache.delete(styleCache.keys().next().value)
```